### PR TITLE
[W4.4b][F09-A2]Ewald Hew

### PR DESCRIPTION
**Pull request by ewaldhew**

### CODE REVIEW

File: src/seedu/addressbook/data/person/Block.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package seedu.addressbook.data.person;
+
+public class Block {
```
Comment (k-l-a) [inline src/seedu/addressbook/data/person/Block.java]: Classes should have javadoc header comments, to comply with the coding standard.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -38,19 +58,22 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block.value
```
Comment (k-l-a) [inline src/seedu/addressbook/data/person/Address.java]: Maybe use get methods instead?

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    public final String value;
```
Comment (k-l-a) [inline src/seedu/addressbook/data/person/Block.java]: Can this variable be private to increase encapsulation?

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -2,17 +2,32 @@
...
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, some street, #01-02, 654321";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Addresses are comma-separated fields consisting of the "
+            + "following: BLOCK, STREET, UNIT, POSTAL_CODE\n"
+            + " BLOCK is a number with optional letter suffix,\n"
+            + " STREET is a string not containing commas,\n"
+            + " UNIT is a number,\n"
+            + " POSTAL_CODE is a six-digit number.";
+    public static final String ADDRESS_VALIDATION_REGEX = "^(?<block>[\\d\\w]\\d*\\w*)?(?:, | )?"
```
Comment (k-l-a) [inline src/seedu/addressbook/data/person/Address.java:28]: Remember to update the documentation (user guide, help, etc) and tests as well when you modify the behavior.

In particular, you should update the user guide to let the readers know that address need to be in a particular format.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -2,17 +2,32 @@
...
+            + "(?<street>[^\\d]+)(?:, |$)"
+            + "(?<unit>[#0-9-]+)?(?:, )?"
+            + "(?<postcode>\\d{6})?";
 
-    public final String value;
+    private static final Pattern ADDRESS_FORMAT = Pattern.compile(ADDRESS_VALIDATION_REGEX);
+    private final Block block;
+    private final Street street;
+    private final Unit unit;
+    private final PostalCode postalCode;
```
Comment (k-l-a) [inline src/seedu/addressbook/data/person/Address.java]: 👍 for making these private. 
However, is it very useful to have these variables if they can't be accessed independently?

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -2,17 +2,32 @@
...
+            + " STREET is a string not containing commas,\n"
+            + " UNIT is a number,\n"
+            + " POSTAL_CODE is a six-digit number.";
+    public static final String ADDRESS_VALIDATION_REGEX = "^(?<block>[\\d\\w]\\d*\\w*)?(?:, | )?"
+            + "(?<street>[^\\d]+)(?:, |$)"
+            + "(?<unit>[#0-9-]+)?(?:, )?"
+            + "(?<postcode>\\d{6})?";
 
-    public final String value;
+    private static final Pattern ADDRESS_FORMAT = Pattern.compile(ADDRESS_VALIDATION_REGEX);
```
Comment (k-l-a) [inline src/seedu/addressbook/data/person/Address.java]: This should be grouped together with the static variables.

**Review by k-l-a — COMMENTED** [review]
@ewaldhew Some comments added. Please close the PR after reading comments.

**Comment by ewaldhew** [comment]
Thanks for the comments! I have just pushed some changes accordingly, hopefully everything is better now.

**Comment by kokonguyen191** [comment]
Close it lol. Also, it's better to not use periods in a commit message.

**Pull request closed without merging**